### PR TITLE
Toolkit migration 2

### PR DESCRIPTION
**Pull request by jamesr101**
## Migrating n-newsletter-signup to toolkit

There are a few things going on here, so please not the following:

- **Pa11y Test**
In the original code, the Make file contained a script for `make test` which verified a11y. The way this worked was by spinning up the demo server and then run a pa11y script against this server. This isn't a "toolkit" way, and wasn't easily translated to the toolkit method of working. Therefore, I have removed the pa11y stuff from the demo server and instead added a Jest test suite. In this Jest test suite, there is a single test which passes the two templates against a Pa11y test called `axe`. This should maintain parity between the original and new way of working.

- **funky sym-linking**
In the original Make file there was the following code when you ran the demo server. This added the templates files into the node_modules:
```
	@rm -rf node_modules/@financial-times/n-newsletter-signup
	@mkdir node_modules/@financial-times/n-newsletter-signup
	@cp -r templates node_modules/@financial-times/n-newsletter-signup/templates/
```
This was a funky way of self-sym-linking, in order for the templates to be able to resolve relative paths such as `n-newsletter-signup/templates/partials/newsletter-form`. Again, this is something that was a bit odd and didn't translate easily to Toolkit. I have been able replace this instead with a function in the `demo/app.js` which registers the handlebar partials with the path names, so it knows where to look without any funky sym-linking shenanigans. (this is just for the demo server)

- **Sass**
In the original code, when you built the demo server, it compiled the scss using `sass`. I have replaced so it now compiles the styles webpack. This is also in line with how n-magnet works. (this is just for the demo server)

[Ticket 🎟 ACC-2354](https://financialtimes.atlassian.net/browse/ACC-2354?atlOrigin=eyJpIjoiMWNkNDU5MmNiODVjNGVmZmIwZjQ3MDA5ZTBhNTBlZTciLCJwIjoiaiJ9)

### Testing:
In order to confirm the changes work and haven't broken anything, I have confirmed the following:
- The demo still works
- The pipeline is passing
- I have locally sym-linked this with n-magnet and can confirm the n-magnet demo is still working (which shows the newsletter signup in action)

### CODE REVIEW

File: .toolkitrc.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  build:local:
+    - WebpackDevelopment
+  run:local:
+    - Node
+options:
+  "@dotcom-tool-kit/circleci":
+    nodeVersion: 16.14.0
+  "@dotcom-tool-kit/vault":
+    team: "next"
+    app: "next-newsletter-signup"
```
Comment (AniaMakes) [inline .toolkitrc.yml:18]: [Observation] Vault has both `next-newsletter-signup` and `n-newsletter-signup`. Are we sure it's the `next-` that we need here when the repo in `n-`?

File: __tests__/templates.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const { toHaveNoViolations, configureAxe } = require('jest-axe');
```
Comment (AniaMakes) [inline __tests__/templates.test.js:1]: [Praise] neat solution!

File: .toolkitrc.yml (hunk 1 above)
Comment (jamesr101) [reply .toolkitrc.yml:18]: There is nothing in `n-newsletter-signup`. It is basically empty. 
```
//teams/next/n-newsletter-signup/continuous-integration
{
  "circleci": {
    "preset": "component"
  },
  "env": {}
}
```

```
//teams/next/n-newsletter-signup/meta
{
  "keys": {},
  "schemaVersion": 1,
  "team": "accounts-customer-products"
}
```

Comment (jamesr101) [reply .toolkitrc.yml:18]: So from that, I assumed it must be "next-newsletter-signup"

Comment (AniaMakes) [reply .toolkitrc.yml:18]: `next-newsletter-signup` (https://github.com/Financial-Times/next-newsletter-signup) looks to be "A sign-up page for FT newsletters that are in front of the paywall." so sounds like a completely different thing. I think Andy Gout is working on it, may be worth a quick word with him.

**Review by AniaMakes — APPROVED** [review]

**Review by joelcarr — APPROVED** [review]
Lovely!

**Pull request merged**